modules/custom/dkan_sql_endpoint/src/Helper/GetStringsFromStateMachineExecution.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\dkan_sql_endpoint\Helper;
+
+/**
+ * Class GetStringsFromStateMachineExecution.
+ */
+class GetStringsFromStateMachineExecution {
+  private $execution;
+  private $strings = [];
+  private $currentString = "";
+
+  /**
+   * Constructor.
+   */
+  public function __construct(array $stateMachineExecution) {
+    $this->execution = $stateMachineExecution;
+  }
+
+  /**
+   * Get.
+   */
+  public function get() {
+    foreach ($this->execution as $states_or_input) {
+      if ($this->isStates($states_or_input)) {
+        $this->processStates($states_or_input);
+        continue;
+      }
+
+      $input = $states_or_input;
+      $this->currentString .= $input;
+    }
+
+    $this->saveAndResetCurrentString();
+
+    return $this->strings;
+  }
+
+  /**
+   * Private.
+   */
+  private function processStates(array $states) {
+    if ($this->containsFirstState($states)) {
+      $this->saveAndResetCurrentString();
+    }
+  }
+
+  /**
+   * Private.
+   */
+  private function saveAndResetCurrentString() {
+    if (!empty($this->currentString)) {
+      $this->strings[] = $this->currentString;
+      $this->currentString = "";
+    }
+  }
+
+  /**
+   * Private.
+   */
+  private function isStates($input): bool {
+    if (!is_array($input)) {
+      return FALSE;
+    }
+
+    return TRUE;
+  }
```

Comment:
How is the above function different from:
`return is_array($input);`

Reply:
It isn't!!! will change to make it more concise.